server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalValueCount.java

```diff
@@ -63,12 +63,20 @@ public double value() {
     }
 
     @Override
-    public InternalAggregation reduce(List<InternalAggregation> aggregations, AggregationReduceContext reduceContext) {
-        long valueCount = 0;
-        for (InternalAggregation aggregation : aggregations) {
-            valueCount += ((InternalValueCount) aggregation).value;
-        }
-        return new InternalValueCount(name, valueCount, getMetadata());
+    protected AggregatorReducer getLeaderReducer(AggregationReduceContext reduceContext, int size) {
+        return new AggregatorReducer() {
```

Comment:
Making these all anonymous classes can make debugging a real pain. I think it's fine because these won't live long, but I could see us wanting to add `toString` implementations on these that help with debugging when we're debugging some issue. Later. No need now.

Reply:
yeah and the capture the oter class which is not nice but it was the best way to move forward. In. follow up we should make all this classes static!